### DISCUSSION TITLE
GitHub Copilot neovim support for NodeJs 18

### DISCUSSION THREAD

**Discussion by alex289** [community/community#38925] [Copilot Conversations] [answered]
### Select Topic Area

Question

### Body

Hey, I really like GitHub copilot but I want to use NodeJs 18 which isn't supported yet. When can we expect it to be supported?

**Comment by cmp-nct** [comment]
Just run copilot on node 16 and use 18 for yourself. No disadvantages there.

**Reply by alex289** [reply]
I know but I don't like using different node versions for different things and since node 18 is the current lts I don't see a reason to not support it

**Comment by tilmann1** [comment, marked as answer]
It's something that the Copilot team has on the radar but there's no clear timeline yet for adding Node.js 18 support. As @cmp-nct mentioned, running on Node.js 16 is the recommended approach :)

**Reply by stevematney** [reply]
I briefly discussed this on Twitter with Tim Pope. The vim plugin is the only one of the copilot plugins that does not ship with its own version of node. This means that I personally have to run a version behind LTS so Copilot will work in vim for me. I can use nvm to easily manage which node version I'm running, which makes it fairly trivial to set my shell to a certain node version. I'd rather run LTS on my system, and somehow configure vim to use nvm to run its shell with its own version of node, but with the way vim runs its shell (in "dumb" mode (that's the name of the mode)), I cannot very easily specify that vim should run with a different version of node than that of my underlying system.

Has there been discussion around shipping the vim plugin with its own version of node (possibly with something like a python virtualenv)? I understand that this may add complexity to the plugin itself, but as it is, it seems that the current decision is to shift the complexity downstream, which seems worse to me.
